statuspage/resource_component.go

```diff
@@ -22,17 +22,27 @@ func resourceComponentCreate(d *schema.ResourceData, m interface{}) error {
 	onlyShowIfDegraded := d.Get("only_show_if_degraded").(bool)
 	status := d.Get("status").(string)
 	showcase := d.Get("showcase").(bool)
-
-	component, err := sp.CreateComponent(
-		client, d.Get("page_id").(string),
-		&sp.Component{
+	groupId := d.Get("group_id").(string)
```

Comment:
Would it cause can issue if empty `group_id` is passed? I think the json marshaller would simply omit if it's empty

Reply:
Yes @smitthakkar96 , it was causing during testing locally, when empty group_id was passed, so added a if condition below to counter it

Reply:
do we need a test case for the empty group_id scenario?

Reply:
> it was causing issue during testing locally when empty group_id was passed
https://github.com/yannh/terraform-provider-statuspage/pull/43/files#diff-09644ef998d4a5dd5e85566bb231048bcb5e8774b92446d61f1bdc5871b24ca5R195 you tried setting `nil` here?

Reply:
never mind you will get
```
interface conversion: interface {} is nil, not string
```